src/index.ts

```diff
@@ -96,6 +100,12 @@ const styled =
           }
         }
 
+        omitProps.forEach((excludeProp) => {
```

Comment:
nit: It's optional. Since restProps is an obj created as a local variable within the render function so directly modifying it is unlikely to cause side effects. But maintaining immutability can help with debugging later or clarify the intention of the code. For example, instead of modifying props directly, creating a new obj can reduce unintended side effects, WDYT?
```suggestion
        const filteredProps = Object.fromEntries(
  Object.entries(restProps).filter(([key]) => !omitProps.includes(key))
)
```

Reply:
I agree generally better to avoid mutation but as correctly mentioned in this case

> it is unlikely to cause side effects

hence not sure how much applicable last motivation:

> creating a new obj can reduce unintended side effect

Besides, I think suggested changes can affect performance. New object is created by iterating all the entries of restProps unconditionally in the render function which in most cases is redundant because withTheme is barely used now and even if used commonly, iterating all props would be excess. 

benchmark confirms it

```
new object |  mutation
           |          
0.02955    |  0.02914 
0.02963    |  0.02899 
0.02946    |  0.02926 
```